src/Symfony/Component/Routing/Generator/Dumper/PhpGeneratorDumper.php

```diff
@@ -38,6 +38,45 @@ public function dump(array $options = array())
             'base_class' => 'Symfony\\Component\\Routing\\Generator\\UrlGenerator',
         ), $options);
 
+        $routes = $this->generateDeclaredRoutes();
+
+        // Fix for https://bugs.php.net/bug.php?id=68057 - Bug #68057 Incorrect parsing of big arrays in PHP 5.6.0
+        if (PHP_VERSION_ID >= 50600 && PHP_VERSION_ID < 50700 && $this->getRoutes()->count() > 32769) {
+            return <<<EOF
+<?php
+
+use Symfony\Component\Routing\RequestContext;
+use Symfony\Component\Routing\Exception\RouteNotFoundException;
+use Psr\Log\LoggerInterface;
+
+/**
+ * {$options['class']}
+ *
+ * This class has been auto-generated
+ * by the Symfony Routing Component.
+ */
+class {$options['class']} extends {$options['base_class']}
+{
+    private static \$declaredRoutes;
+
+    /**
+     * Constructor.
+     */
+    public function __construct(RequestContext \$context, LoggerInterface \$logger = null)
+    {
+        \$this->context = \$context;
+        \$this->logger = \$logger;
+        if ( null === self::\$declaredRoutes ) {
+            self::\$declaredRoutes = {$routes};
```

Comment:
shouldn't we just make this the default implementation and remove the duplication?
I don't think there would we any significant performance difference